packages/http/src/responses.ts

```diff
@@ -31,7 +32,7 @@ export function getResponsesForOperation(
   const diagnostics = createDiagnosticCollector();
   const responseType = operation.returnType;
   const responses = new ResponseIndex();
-  if (responseType.kind === "Union") {
+  if (responseType.kind === "Union" && !getDiscriminatedUnion(program, responseType)[0]) {
```

Comment:
Also considered adding an `isDiscriminatedUnion` function in the compiler so we don't have to unnecessarily build the union.

Reply:
there might be a typekit for that

Reply:
Not quite for `isDiscriminatedUnion`, but I did update to use the typekit equivalents.